src/main/java/com/google/codeu/servlets/CinemaInitServlet.java

```diff
@@ -0,0 +1,67 @@
+package com.google.codeu.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.codeu.data.Datastore;
+import com.google.codeu.data.Marker;
+import java.io.IOException;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonElement;
+import com.google.codeu.utils.CsvLoader;
+
+/** Handles initialization of {@link Marker} instances for each cinema. */
+@WebServlet("/cinema-init")
+public class CinemaInitServlet extends HttpServlet {
+
+  private static String cinemaLocationFilePath = "/WEB-INF/cinemaLocations.csv";
+  private Datastore datastore;
+  private JsonArray cinemaInfos;
+
+  @Override
+  public void init() {
+    datastore = new Datastore();
+  }
+
+  /**
+   * Checks user login credentials. If user is admin, loads cinema location and description from
+   * csv file to datastore. If not, redirect the user to home page.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    // Redirects user if not logged in
+    if (!userService.isUserLoggedIn()) {
```

Comment:
nit: this if branch can be combined with the "not admin" if branch below

Reply:
I think will be changed to something like if not logged in, prompt for login. If not admin, send redirection, so still need 2 separate branches